Source/WebGPU/WebGPU/RenderPipeline.mm

```diff
@@ -478,13 +640,36 @@ static void populateStencilOperation(MTLStencilDescriptor *mtlStencil, const WGP
     auto mtlFrontFace = frontFace(descriptor.primitive.frontFace);
     auto mtlCullMode = cullMode(descriptor.primitive.cullMode);
 
-    MTLRenderPipelineReflection *reflection;
     NSError *error = nil;
-    id<MTLRenderPipelineState> renderPipelineState = [m_device newRenderPipelineStateWithDescriptor:mtlRenderPipelineDescriptor options:pipelineLayout ? MTLPipelineOptionNone : MTLPipelineOptionArgumentInfo reflection:&reflection error:&error];
+    id<MTLRenderPipelineState> renderPipelineState = [m_device newRenderPipelineStateWithDescriptor:mtlRenderPipelineDescriptor error:&error];
     if (error || !renderPipelineState)
         return RenderPipeline::createInvalid(*this);
 
-    return RenderPipeline::create(renderPipelineState, mtlPrimitiveType, mtlIndexType, mtlFrontFace, mtlCullMode, mtlDepthClipMode, depthStencilDescriptor, reflection, pipelineLayout, *this);
+    if (!pipelineLayout) {
```

Comment:
Seems like a lot of this code could be shared with ComputePipeline?

Reply:
I will pull out a helper member function into `Device` to minimize the duplication.